### ISSUE TITLE
Utility to create sim from a config file.

### ISSUE THREAD

**Issue by fpagliughi** [warthog618/gpiosim-rs#1]
It would be pretty cool to have a little, optional, utility app that could take a description of a desired simulation (maybe a TOML config file) and create it on the host.

Then, as an example provide config files that create the GPIO setup for a common make board, like an RPi or BeagleBone Black.

Does anything like this already exist? If not, I can try to throw something together.

**Comment by fpagliughi** [comment]
Oh. I was proposing something that would create a persistent simulation, which would still exist after the utility is run... like how `basic_sim.sh` works. I didn't realize that `Sim` tears down the simulator when dropped.

I wonder if we couldn't add a `persist: bool` flag to `Sim` such that drop would become:
```
impl Drop for Sim {
    fn drop(&mut self) {
        if !self.persist {
            self.cleanup_configfs();
        }
    }
}
```

I'm thinking if you're developing some code for an embedded Linux board, it would be nice to have the simulated GPIO space for that board on your laptop for an extended period, and it would be nice to define it with a relatively simple config file. I think that having the simulation stay in place would be nicer than having to start it every time you want to debug.

**Comment by warthog618** [comment]
Interesting idea, but is a very different use case.  The Sim is intended to be used to dynamically create gpiochips for testing, to allow access to the chip from the kernel side during the test and to cleanup when the test is complete.  As each Sim is totally independent, test cases can be run in parallel without interfering with each other and without any additional use space locking/serialisation.  Persistent sims are contrary to that approach and open multiple cans of worms, including the aforementioned locking, and the mapping from gpiochip to the gpio-sim in sysfs space which is readily determined during gpio-sim creation, but is tedious to determine afterwords.

Note that the gpio-sims can be created via [device-tree config](https://www.kernel.org/doc/html/latest/admin-guide/gpio/gpio-sim.html), so the approach I would use to emulate a particular platform is to create a corresponding device tree overlay.  Either that or just write a script like `basic_sim.sh`, or a mini custom app in Rust, to create them. Either would be preferable to creating yet another format that describes the gpio-sim/GPIO pinouts.

Being able to create and/or bind to existing gpio-sims is certainly doable if you do want to drive them from Rust, but not sure if Sim is the place for it - either a separate class or even a totally separate library.
